File: frontend/src/api/promptBuilder.ts
import { apiFetch } from './client';

export interface PromptBuilderRequest {
  run_id: string;
  user_question: string;
  helper_model_config_id?: string;
}

export interface PromptBuilderResponse {
  reasoning: string;
  suggestion: 'improve_prompt' | 'no_change' | 'error';
  proposed_prompt?: string;
  explanation?: string;
  error_message?: string;
}

export interface ApproveChangeRequest {
  run_id: string;
  proposed_prompt: string;
  explanation: string;
}

export interface ApproveChangeResponse {
  version_id: string;
  version_number: number;
  new_run_id: string;
  new_run_output?: string;
  error_message?: string;
}

export const promptBuilderApi = {
  askHelper: (projectId: string, request: PromptBuilderRequest) =>
    apiFetch<PromptBuilderResponse>(
      `/projects/${projectId}/prompt-builder/ask`,
      { method: 'POST', body: JSON.stringify(request) }
    ),

  approveChange: (projectId: string, request: ApproveChangeRequest) =>
    apiFetch<ApproveChangeResponse>(
      `/projects/${projectId}/prompt-builder/approve-change`,
      { method: 'POST', body: JSON.stringify(request) }
    ),
};

File: backend/app/schemas/prompt_builder.py
from pydantic import BaseModel


class PromptBuilderRequest(BaseModel):
    run_id: str
    user_question: str
    helper_model_config_id: str | None = None


class PromptBuilderResponse(BaseModel):
    reasoning: str
    suggestion: str  # "improve_prompt" | "no_change" | "error"
    proposed_prompt: str | None = None
    explanation: str | None = None
    error_message: str | None = None


class ApproveChangeRequest(BaseModel):
    run_id: str
    proposed_prompt: str
    explanation: str


class ApproveChangeResponse(BaseModel):
    version_id: str
    version_number: int
    new_run_id: str
    new_run_output: str | None
    error_message: str | None = None
